FreeRTOS/FreeRTOS-Cellular-Interface
# Fix MISRA C 2012 deviations

Description
-----------
Fix MISRA C 2012 deviations

In this PR:
* Adding "UNITTEST" and "COV_ANALYSIS" to align FreeRTOS libraries design
* Fix rule 12.3 not to use comma operator
* Fix rule 13.4 not to use the returned value of the assignment operator
* Fix rule 17.7 for unused return value
* Fix rule 18.4 not to use arithmetic on pointer type
* Fix rule 2.2 to remove overwritten initialized value
* Fix forward NULL warning

Test Steps
-----------
Build coverity_analysis project with coverity 2023.6.1 should show no MISRA deviations.

Checklist:
----------



Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
